Ubuntu 18.04 inbox driver failure: ‘struct mlx5dv_cq’ has no member named ‘uar’

```
4.15.0-34-generic #37-Ubuntu SMP Mon Aug 27 15:22:18 UTC 2018 aarch64 aarch64 aarch64 GNU/Linux
```

```
  CC       ib/rc/base/libuct_la-rc_ep.lo
../../../src/uct/ib/mlx5/ib_mlx5.c: In function ‘uct_ib_mlx5_get_cq’:
../../../src/uct/ib/mlx5/ib_mlx5.c:43:53: error: ‘struct mlx5dv_cq’ has no member named ‘uar’
     mlx5_cq->uar       = uct_dv_get_info_uar0(dcq.dv.uar);
                                                     ^
```

--- a/config/m4/ib.m4
+++ b/config/m4/ib.m4
@@ -218,9 +218,7 @@ AS_IF([test "x$with_ib" == xyes],
               AC_CHECK_DECLS([ibv_alloc_td],
                       [has_res_domain=yes], [], [[#include <infiniband/verbs.h>]])])
 
-       AS_IF([test "x$has_res_domain" == xyes], [], [
-               AC_MSG_WARN([Cannot use mlx5 accel because resource domains are not supported])
-               AC_MSG_WARN([Please upgrade MellanoxOFED to 3.1 or above])
+       AS_IF([test "x$has_res_domain" == xyes -a "x$have_cq_io" == xyes ], [], [
                with_mlx5_hw=no])
 
        AS_IF([test "x$with_mlx5_hw" == xyes],

